server/tserver/src/main/java/org/apache/accumulo/tserver/tablet/Scanner.java

```diff
@@ -120,16 +108,15 @@ public ScanBatch read() throws IOException, TabletClosedException {
       }
 
       sawException = true;
-      dataSource.close(true);
       throw ioe;
-    } catch (RuntimeException re) {
+    } catch (Exception e) {
```

Comment:
Careful about catching other RuntimeExceptions and wrapping them again with yet another RuntimeException. Also, try to pick a more specific RuntimeException, if possible. Perhaps `IllegalStateException`, `IllegalArgumentException`, or `UncheckedIOException`.